src/java.base/share/classes/java/io/BufferedOutputStream.java

```diff
@@ -83,14 +128,46 @@ private void flushBuffer() throws IOException {
         }
     }
 
+    /**
+     * Grow buf to fit an additional len bytes if needed.
+     * If possible, it grows by len+1 to avoid flushing when len bytes
+     * are added. A no-op if the buffer is not resizable.
+     */
+    private void growIfNeeded(int len) {
```

Comment:
Hello Alan, like some methods in other classes (BufferedInputStream#fill() for example), should we mention here that this method assumes that it's being called while a lock is held?

Reply:
Okay, we can add a comment to these methods (asserts are possible but tend not to be used in this area).